frame/assets-registry/src/lib.rs

```diff
@@ -250,23 +280,23 @@ pub mod pallet {
 	impl<T: Config> RemoteAssetRegistry for Pallet<T> {
 		type AssetId = T::LocalAssetId;
 
-		type AssetNativeLocation = T::ForeignAssetId;
+		type AssetNativeLocation = T::Location;
```

Comment:
Native has notion of native to some chain or protocol, so all assets are in some sense native. may be call AssetOriginLocation `assets originated from location`